src/sentry/monitors/logic/mark_ok.py

```diff
@@ -59,6 +63,18 @@ def mark_ok(checkin: MonitorCheckIn, ts: datetime):
                         "grouphash": incident.grouphash,
                     },
                 )
+                # if incident was longer than the broken env time, check if there was a broken detection that is also now resolved
+                if incident.starting_timestamp <= timezone.now() - timedelta(
+                    days=NUM_DAYS_BROKEN_PERIOD
+                ):
+                    if incident.monitorenvbrokendetection_set.count() == 1:
```

Comment:
Will there ever be more than one in this set? We may as well use .exists if not

Reply:
oh yes that is nicer for sure